src/common/transport/transport-client.js

```diff
@@ -0,0 +1,80 @@
+import { TransportClientError } from '../errors';
+import rp from 'request-promise';
```

Comment:
The way I've seen it done is you provide a separate look up for a module with the same API see: https://github.com/defunctzombie/package-browser-field-spec. This basically provides context on which transporter to leverage in a given environment. 

As for the Babel/ES5 browser thing I hear you. I don't want to derail this project/pull request would you guys be able to sync up IRL real quick?